### ISSUE TITLE
Shared components for OHDSI web tools

### ISSUE THREAD

**Issue by pavgra** [OHDSI/WebAPI#505]
Today we have several web tools in OHDSI which have been evolving for a significant term on themselves and during their evolution implemented quite similar functionality separately:
![arachne webapi unification as is](https://user-images.githubusercontent.com/10010071/42484057-fd040bd0-83be-11e8-8539-743936aa1385.png)

It would be great to remove the redundancy from the tools by identifying their common parts and moving them into shared components so that different teams wouldn't spend time extra on doing the same work, features would be shared and better code quality would be achieved. It would force real community collaboration. E.g. the following could be done:
![arachne webapi unification to be](https://user-images.githubusercontent.com/10010071/42484148-7cc53eb6-83bf-11e8-8c4c-9f99d1337e0e.png)

I'd suggest starting with the most obvious parts such as Data Source management and move forward with others as things get clearer and the design is polished and approved by the community.

Tagging @chrisknoll , @pbr6cornell , @gklebanov

**Comment by konstjar** [comment]
One more suggestion. Can we automate ACHILLES generation from ATLAS Configuration screen? Use cases:
- data set was updated and there is a need to re-run ACHILLES 
- new data set was added and ACHILLES should be executed

**Comment by pavgra** [comment]
@konstjar, good point, the functionality should be presented in the shared DS management & healthcheck component (and since the Achilles generation button is already available in Arachne, we'll just take it from there)

**Comment by t-abdul-basser** [comment]
Seconded. This would be very useful feature.
